### PR TITLE
stat factory cleanup & comments

### PR DESCRIPTION
**Pull request by nev-r**
this should be a complete no-op, i just wanted to:
- add comments
- rename a couple functions to better express how they mutate their inputs
- rename some too-generic variables
- use early-on `continues` in for loops, to avoid unnecessary indentation and long brace nesting
- remove unnecessary optional chains

### CODE REVIEW

File: src/app/inventory/store/stats.ts [hunk 1]
```diff
@@ -158,7 +162,8 @@ export function buildStats(
 }
 
 /**
- * This builds a class items stat values from its masterwork and mod sockets.
+ * This builds a class item's stat values from its masterwork and mod sockets.
+ * to-do: can we get rid of this once we aren't using Live Stats for armor?
```
Comment (ryan-rushton) [inline src/app/inventory/store/stats.ts:166]: Yes,  the whole reason this is here is because we know what the base stat values are (spoiler it's 0)

**Review by ryan-rushton — APPROVED** [review]
Didn't test but this looks good to me.

File: src/app/inventory/store/stats.ts [hunk 2]
```diff
@@ -279,6 +289,11 @@ function buildInvestmentStats(
   return ret;
 }
 
+/**
+ * builds and returns a single DimStat, using InvestmentStat information,
+ * stat def, statgroup def, and the item's StatDisplayDefinition,
+ * which determins which stats are displayed and how they are interpolated
```
Comment (bhollis) [inline src/app/inventory/store/stats.ts:295]:
```suggestion
 * which determines which stats are displayed and how they are interpolated
```

File: src/app/inventory/store/stats.ts [hunk 3]
```diff
@@ -334,48 +355,57 @@ function enhanceStatsWithPlugs(
     return;
   }
 
+  const existingStatsByHash: NodeJS.Dict<DimStat> = _.keyBy(existingStats, (s) => s.statHash);
```
Comment (bhollis) [inline src/app/inventory/store/stats.ts:358]: `NodeJS.Dict`? Maybe `_.Dictionary` instead? Though typing this is generally unnecessary.

File: src/app/inventory/store/stats.ts [hunk 4]
```diff
@@ -389,24 +419,24 @@ function enhanceStatsWithPlugs(
...
- * will contribute to the items stats when socketed.
+ * Generates the stat modification map for each DimPlug in a DimSocket
+ * and attaches it to the DimPlug's stats property
  */
-function buildPlugStats(
+function attachPlugStats(
   socket: DimSocket,
-  statsByHash: { [statHash: number]: DimStat },
-  statDisplays: { [statHash: number]: DestinyStatDisplayDefinition }
+  statsByHash: NodeJS.Dict<DimStat>,
```
Comment (bhollis) [inline src/app/inventory/store/stats.ts:432]: I think I actually don't love these types - it misses the built-in documentation of what the key type is and what it's for. The old type is more verbose but it's clearly indexed on statHash.

**Review by bhollis — APPROVED** [review]

File: src/app/inventory/store/stats.ts (hunk 4 above)
Comment (nev-r) [reply src/app/inventory/store/stats.ts:432]: i'll swap it back over to named keys, but the NodeJS.Dict was to make the keys optional so later lookups are properly `| undefined`

**Pull request merged**